src/authorization/SimpleAclAuthorizer.ts

```diff
@@ -0,0 +1,143 @@
+  private readonly aclManager: AclManager;
+  private readonly containerManager: ContainerManager;
+  private readonly store: ResourceStore;
+
+  public constructor(aclManager: AclManager, containerManager: ContainerManager, store: ResourceStore) {
+    super();
+    this.aclManager = aclManager;
+    this.containerManager = containerManager;
+    this.store = store;
+  }
+
+  public async canHandle(): Promise<void> {
+    // Can handle everything
+  }
+
+  /**
+   * Checks if an agent is allowed to execute the requested actions.
+   * Will throw an error if this is not the case.
+   * @param input - Relevant data needed to check if access can be granted.
+   */
+  public async handle(input: AuthorizerArgs): Promise<void> {
+    const store = await this.getAclRecursive(input.identifier);
+    if (await this.aclManager.isAcl(input.identifier)) {
+      this.checkPermission(input.credentials, store, 'control');
+    } else {
+      (Object.keys(input.permissions) as (keyof PermissionSet)[]).forEach((key): void => {
+        if (input.permissions[key]) {
+          this.checkPermission(input.credentials, store, key);
+        }
+      });
+    }
+  }
+
+  /**
+   * Checks if any of the triples in the store grant the agent permission to use the given mode.
+   * Throws a {@link ForbiddenHttpError} or {@link UnauthorizedHttpError} depending on the credentials
+   * if access is not allowed.
+   * @param agent - Agent that wants access.
+   * @param store - A store containing the relevant triples for authorization.
+   * @param mode - Which mode is requested. Probable one of ('write' | 'read' | 'append' | 'control').
+   */
+  private checkPermission(agent: Credentials, store: Store, mode: string): void {
+    const modeString = `${prefix}${mode[0].toUpperCase()}${mode.slice(1).toLowerCase()}`;
+    const auths = store.getQuads(null, `${prefix}mode`, modeString, null).map((quad: Quad): Term => quad.subject);
+    if (!auths.some((term): boolean => this.hasAccess(agent, term, store))) {
+      throw agent.webID ? new ForbiddenHttpError() : new UnauthorizedHttpError();
+    }
+  }
+
+  /**
+   * Checks if the given agent has access to the modes specified by the given authorization.
+   * @param agent - Credentials of agent that needs access.
+   * @param auth - acl:Authorization that needs to be checked.
+   * @param store - A store containing the relevant triples of the authorization.
+   *
+   * @returns If the agent has access.
+   */
+  private hasAccess(agent: Credentials, auth: Term, store: Store): boolean {
+    if (store.countQuads(auth, `${prefix}agentClass`, 'http://xmlns.com/foaf/0.1/Agent', null) > 0) {
+      return true;
+    }
+    if (!agent.webID) {
+      return false;
+    }
+    if (store.countQuads(auth, `${prefix}agentClass`, 'http://xmlns.com/foaf/0.1/AuthenticatedAgent', null) > 0) {
+      return true;
+    }
+    return store.countQuads(auth, `${prefix}agent`, agent.webID, null) > 0;
```

Comment:
Is there no `exists` check instead of counting? But works for me.

Reply:
Unfortunately [not](https://github.com/rdfjs/N3.js#searching-quads-or-entities). Closest would be using a `some` call, but code would be a bit messier and these data sets are going to be tiny anyway.

Reply:
Fair enough.